Automattic/wp-calypso
# Gutenboarding: add domain availability check

#### Background

This check is being split into two separate PRs.

1. Setup the selectors for the domain availability check.
2. Use these selectors on the UI.

The second PR hasn’t been opened yet but these selectors would be used in the following scenarios:

**Scenario 1:**
A user enters a FQDN for their site name. E.g.

* “Automattic” - Check won’t be performed.
* “Automattic.com” - Check will be performed.

**Scenario 2:**
A user selects a domain that is displayed as a suggested domain.

**Scenario 3:**
In the case the user types a FQDN in the search bar on `/new/domains`, we’d check the availability of that domain specifically as soon as the user finished typing

There’s more on this here https://github.com/Automattic/wp-calypso/issues/40971#issuecomment-614828121

> Basically you have to assume that the suggestion endpoint is not always 100% accurate and might return unavailable domains. So, once a domain is selected, you should for it’s availability via the is-available endpoint.

#### Changes proposed in this Pull Request

* Adding domain availability selector to the `domain-suggestions` package.
* If the API request fails, we will return a status of `'unknown'` which is what the API does if it can not accurately determine the availability status.

#### Testing instructions

These changes are not yet being used, so we would be best checking the absence of them and that everything else around domain suggestions runs as normal.

* Go to [/new](https://hash-fb5de047678ac106a94e2d36f47131b5ac43f203.calypso.live/new) and enter a site name
* Check the network tab that a request is not being made to the `/is-available` API
* Check that the domain suggestions get returned as normal and you've able to proceed through the process.

Partially implements changes requested by #41220
